{
  "id": "CVE-2017-1000158",
  "package_name": "python",
  "fix_state": "unknown",
  "cpes": [
    "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
  ],
  "version_constraint": "< 2.7.15 || >= 3.4.0, < 3.4.8 || >= 3.5.0, < 3.5.5",
  "fixed_in_versions": "null",
  "severity": "Critical",
  "description": "CPython (aka Python) up to 2.7.13 is vulnerable to an integer overflow in the PyString_DecodeEscape function in stringobject.c, resulting in heap-based buffer overflow (and possible arbitrary code execution)",
  "urls": [
    "http://www.securitytracker.com/id/1039890",
    "https://bugs.python.org/issue30657",
    "https://lists.debian.org/debian-lts-announce/2017/11/msg00035.html",
    "https://lists.debian.org/debian-lts-announce/2017/11/msg00036.html",
    "https://lists.debian.org/debian-lts-announce/2018/09/msg00030.html",
    "https://lists.debian.org/debian-lts-announce/2018/09/msg00031.html",
    "https://security.gentoo.org/glsa/201805-02",
    "https://security.netapp.com/advisory/ntap-20230216-0001/",
    "https://www.debian.org/security/2018/dsa-4307"
  ]
}
